components/authz-service/server/v2/project.go

```diff
@@ -401,13 +401,17 @@ func (s *ProjectState) UpdateRule(ctx context.Context, req *api.UpdateRuleReq) (
 	}
 
 	resp, err := s.store.UpdateRule(ctx, r)
-	if err != nil {
-		if err == storage_errors.ErrNotFound {
-			return nil, status.Errorf(codes.NotFound, "rule with ID %q not found", req.Id)
-		}
-		if err == storage_errors.ErrChangeProjectForRule {
-			return nil, status.Errorf(codes.FailedPrecondition,
-				"cannot change project_id for existing rule with ID %q ", req.Id)
+	switch err {
+	case nil: // continue
+	case storage_errors.ErrNotFound:
+		return nil, status.Errorf(codes.NotFound, "rule with ID %q not found", req.Id)
+	case storage_errors.ErrChangeProjectForRule:
+		return nil, status.Errorf(codes.FailedPrecondition,
+			"cannot change project_id for existing rule with ID %q ", req.Id)
+	default:
+		switch err.(type) {
+		case *storage_errors.ForeignKeyError:
+			return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
It's because we've been mixing error constants and error types. The types of error constants are not informative, and can't be used in a `switch err.(type)` construct.

Yes, turning our error constants into error types would help, but
1. what we have here doesn't hurt _that much_
2. if we do it, let's do it in a separate PR (the nested construct here exists in other places, too)